nullaway/src/main/java/com/uber/nullaway/LibraryModels.java

```diff
@@ -94,92 +92,80 @@
    */
   final class MethodRef {
 
-    public final String clazz;
-    public final String methodSig;
-
-    private MethodRef(String clazz, String methodSig) {
-      this.clazz = clazz;
-      this.methodSig = methodSig;
+    public final String enclosingClass;
+    public final String methodName;
+    public final String methodArgs;
+    @Nullable public final String genericArgs;
+
+    private MethodRef(
+        String enclosingClass, String methodName, String methodArgs, @Nullable String genericArgs) {
+      this.enclosingClass = enclosingClass;
+      this.methodName = methodName;
+      this.methodArgs = methodArgs;
+      this.genericArgs = genericArgs;
     }
 
     /**
-     * @param clazz containing class
-     * @param methodSig method signature in the appropriate format (see class docs)
+     * @param enclosingClass containing class
+     * @param methodArgs method signature in the appropriate format (see class docs)
      * @return corresponding {@link MethodRef}
      */
-    public static MethodRef methodRef(Class<?> clazz, String methodSig) {
-      return methodRef(clazz.getName(), methodSig);
+    public static MethodRef methodRef(String enclosingClass, String methodName, String methodArgs) {
```

Comment:
This is a breaking API change.  I realized I can mostly avoid it by just parsing the method signatures getting passed in before, rather than forcing callers to pass the parts of a signature separately.  Perf shouldn't be a concern as this happens once at initialization time.  Will update the PR to undo the API change.